src/build/arg/mod.rs

```diff
@@ -2581,22 +2582,44 @@ impl<'help> Arg<'help> {
     ///         .long("opt"))
     ///     .arg(Arg::new("other")
     ///         .long("other")
-    ///         .default_value_if("opt", Some("special"), "default"))
+    ///         .default_value_if("opt", Some("special"), Some("default")))
     ///     .get_matches_from(vec![
     ///         "prog", "--opt", "hahaha"
     ///     ]);
     ///
     /// assert_eq!(m.value_of("other"), None);
     /// ```
+    ///
+    /// We can also remove a default if the flag was set.
+    ///
+    /// ```rust
+    /// # use clap::{App, Arg};
+    /// let m = App::new("prog")
+    ///     .arg(Arg::new("flag")
+    ///         .long("flag"))
+    ///     .arg(Arg::new("other")
+    ///         .long("other")
+    ///         .default_value("default")
+    ///         .default_value_if("flag", None, None))
+    ///     .get_matches_from(vec![
+    ///         "prog", "--flag"
+    ///     ]);
+    ///
+    /// assert_eq!(m.value_of("other"), None);
+    /// ```
     /// [`Arg::takes_value(true)`]: ./struct.Arg.html#method.takes_value
     /// [`Arg::default_value`]: ./struct.Arg.html#method.default_value
     pub fn default_value_if<T: Key>(
         self,
         arg_id: T,
         val: Option<&'help str>,
-        default: &'help str,
+        default: impl Into<Option<&'help str>>,
```

Comment:
Please use a type parameter instead of writing `impl` here. Check other functions in this file where we have `Into`.

Reply:
Changed.